src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -1386,13 +1389,15 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 			this._localStore.add(this._webview.webview.onDidBlur(() => {
 				this._outputFocus.set(false);
 				this._webviewFocused = false;
+				this._webviewHasFocus.set(false);
```

Comment:
A more general context key like this is more useful, but then it shouldn't be managed at the individual widget level. It won't behave as expected that way (other webviews won't trigger this) and will likely start conflicting between widgets

Reply:
> It won't behave as expected that way (other webviews won't trigger this) and will likely start conflicting between widgets

What do you propose?

Reply:
Is the suggestion to add something more generic into the core code that creates webviews?

Reply:
@amunger  I have no idea how to do this, open to suggestions.

Reply:
@mattbierner Perhaps you can help here.

Reply:
I thought of updating the context key in the `WebviewService` class in `src/vs/workbench/contrib/webview/browser/webviewService.ts`

For that, I'd need to pass an optional parameter of `IContextKeyService` as follows
```typescript
createWebviewElement(initInfo: WebviewInitInfo, contextKeyService?: IContextKeyService): IWebviewElement;
```

Reply:
Never mind, found that webViewService has already got access to the contextService.